src/containers/Article-new.js

```diff
@@ -0,0 +1,617 @@
+  }
+
+  const handleScroll = _.debounce(() => {
+    const scrollHeight = window.scrollY || document.documentElement.scrollTop
+    const totalHeight = document.documentElement.scrollHeight
+    const targetHeight =
+      totalHeight * articleReadCountConditionConfig.reading_height
+    if (scrollHeight >= targetHeight) {
+      setIsReachedArticleReadTargetHeight(true)
+    }
+  }, 500)
+
+  const handleVisibilityChange = _.throttle(() => {
+    if (document.visibilityState === 'hidden') {
+      if (isActive) {
+        if (inactiveTimerId.current) {
+          window.clearTimeout(inactiveTimerId.current)
+        }
+        calculateActiveTime()
+        if (activeTime >= articleReadTimeConditionConfig.min_active_time) {
+          sendActiveTime()
+        }
+        isActive = false
+      }
+    } else {
+      if (!isActive) {
+        isActive = true
+        startReadingTime = Date.now()
+      }
+      startInactiveTimer()
+    }
+  }, 1000)
+
+  const handleUserActivity = _.debounce(() => {
+    if (!isActive) {
+      isActive = true
+      startReadingTime = Date.now()
+    }
+    startInactiveTimer()
+  }, 500)
+
+  const handlePagehide = () => {
+    calculateActiveTime()
+    if (activeTime >= articleReadTimeConditionConfig.min_active_time) {
+      sendActiveTime()
+    }
+  }
+
+  useEffect(() => {
+    // reset readingCount state
+    history.block(() => {
+      setIsBeenRead(false)
+      setIsReachedArticleReadTargetHeight(false)
+      setIsReachedArticleReadTargetTime(false)
+      calculateActiveTime()
+      if (activeTime >= articleReadTimeConditionConfig.min_active_time) {
+        sendActiveTime()
+      }
+      return true
+    })
+  }, [history])
+
+  useEffect(() => {
+    fetchAFullPostWithCatch(slugToFetch)
+    sendUserFootprint()
+  }, [slugToFetch])
+
+  useEffect(() => {
+    fetchAFullPostWithCatch(slugToFetch)
+    startReadingCountTimer()
+    startReadingTime = Date.now()
+    document.addEventListener('scroll', handleScroll)
+    document.addEventListener('visibilitychange', handleVisibilityChange)
+    document.addEventListener('mousemove', handleUserActivity)
+    document.addEventListener('click', handleUserActivity)
+    window.addEventListener('pagehide', handlePagehide)
+
+    handleVisibilityChange()
+
+    localForage
+      .getItem(bsConst.keys.fontLevel)
+      .then(value => {
+        if (value !== fontLevel) {
+          changeFontLevel(value)
+        }
+      })
+      .catch(err => {
+        console.warn('Can not get item from browser storage: ', err)
+      })
+
+    return () => {
+      document.removeEventListener('scroll', handleScroll)
+      document.removeEventListener('visibilitychange', handleVisibilityChange)
+      document.removeEventListener('mousemove', handleUserActivity)
+      document.removeEventListener('click', handleUserActivity)
+      window.removeEventListener('pagehide', handlePagehide)
+    }
+  }, [])
+
+  useEffect(() => {
+    if (!isFetchingPost) {
```

Comment:
The default value of `isFetchingPost` is false, which means it's value might change like : false -> true -> false
Do we need to check the previous props value in order to match the false after true ?

Reply:
fixed